packages/react/src/components/TextField/TextField.base.tsx

```diff
@@ -543,38 +547,23 @@ export class TextFieldBase extends React.Component<ITextFieldProps, ITextFieldSt
 
     const element = event.target as HTMLInputElement;
     const value = element.value;
-    // Ignore this event if the value is undefined (in case one of the IE bugs comes back)
-    if (value === undefined || value === this._lastChangeValue) {
+    // Ignore this event if any of the following are true:
+    // - the value is undefined (in case one of the IE bugs comes back)
+    // - it's a duplicate event (important since onInputChange is called twice per actual user event)
+    // - it's the same as the previous value
+    if (value === undefined || value === this._lastChangeValue || value === _getValue(this.props, this.state) || '') {
```

Comment:
With the operator precedence, what it's doing is checking `value` against `_getValue(this.props, this.state) || ''`. If `_getValue` returns something falsy (most likely undefined), it will default to checking against `''`. This part of the logic is roughly the same as before, just moved around.

Reply:
Oops you're right, I'm missing some parentheses